# Data Collection

## Objective

The purpose of this notebook is to collect historical WTI Crude Oil Futures data, inspect the dataset structure, and prepare a reproducible data source for market intelligence analysis.

Dataset:
- Asset: WTI Crude Oil Futures
- Source: Yahoo Finance
- Symbol: CL=F
- Frequency: Daily

In [1]:
import sys
from pathlib import Path

PROJECT_ROOT = Path.cwd().parent
sys.path.append(str(PROJECT_ROOT))

In [2]:
from src.data.loader import load_wti_data
df = load_wti_data()
df.head()

[*********************100%***********************]  1 of 1 completed


Price,Adj Close,Close,High,Low,Open,Volume
Ticker,CL=F,CL=F,CL=F,CL=F,CL=F,CL=F
Date,,,,,,
2021-06-09,69.959999,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,70.910004,71.239998,69.680000,70.099998,563575
2021-06-14,70.879997,70.879997,71.779999,70.650002,70.650002,319639
2021-06-15,72.120003,72.120003,72.489998,70.809998,71.160004,336035


In [3]:
df.columns

MultiIndex([('Adj Close', 'CL=F'),
            (    'Close', 'CL=F'),
            (     'High', 'CL=F'),
            (      'Low', 'CL=F'),
            (     'Open', 'CL=F'),
            (   'Volume', 'CL=F')],
           names=['Price', 'Ticker'])

In [4]:
df.columns = df.columns.get_level_values(0)
df.head()

Price,Adj Close,Close,High,Low,Open,Volume
Date,,,,,,
2021-06-09,69.959999,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,70.910004,71.239998,69.680000,70.099998,563575
2021-06-14,70.879997,70.879997,71.779999,70.650002,70.650002,319639
2021-06-15,72.120003,72.120003,72.489998,70.809998,71.160004,336035


In [5]:
df.columns

Index(['Adj Close', 'Close', 'High', 'Low', 'Open', 'Volume'], dtype='str', name='Price')

In [6]:
(df['Adj Close'] == df['Close']).all()

np.True_

In [7]:
df.drop(columns=['Adj Close'], inplace=True)

In [8]:
df.head()

Price,Close,High,Low,Open,Volume
Date,,,,,
2021-06-09,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,71.239998,69.680000,70.099998,563575
2021-06-14,70.879997,71.779999,70.650002,70.650002,319639
2021-06-15,72.120003,72.489998,70.809998,71.160004,336035


In [10]:
df.tail()

Price,Close,High,Low,Open,Volume
Date,,,,,
2026-06-03,96.019997,97.000000,93.449997,93.449997,260613
2026-06-04,93.040001,95.910004,91.910004,95.750000,219500
2026-06-05,90.540001,93.629997,89.680000,92.820000,252903
2026-06-08,91.300003,95.470001,90.389999,93.000000,278650
2026-06-09,88.879997,90.000000,88.279999,89.400002,18385


In [11]:
df.shape

(1258, 5)

### Data Information

In [12]:
df.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 1258 entries, 2021-06-09 to 2026-06-09
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Close   1258 non-null   float64
 1   High    1258 non-null   float64
 2   Low     1258 non-null   float64
 3   Open    1258 non-null   float64
 4   Volume  1258 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 59.0 KB


### Dataset Description

In [13]:
df.describe()

Price,Close,High,Low,Open,Volume
count,1258.000000,1258.000000,1258.000000,1258.000000,1.258000e+03
mean,78.055151,79.539531,76.576804,78.114149,3.199007e+05
std,13.108203,13.736083,12.483513,13.142783,1.269699e+05
min,55.270000,56.700001,54.980000,55.230000,0.000000e+00
25%,69.242498,70.387499,68.165003,69.332500,2.590525e+05
50%,75.864998,77.315002,74.619999,75.905003,3.157500e+05
75%,83.849998,85.582497,82.557499,84.025002,3.773585e+05
max,123.699997,130.500000,120.790001,124.660004,1.107193e+06


### Reset Index

In [14]:
df.reset_index(inplace=True)

### Save Raw Dataset

In [15]:
output_path = "../data/raw/wti.csv"

df.to_csv(output_path, index=False)

### Verification

In [16]:
import pandas as pd
saved_df = pd.read_csv(output_path)

saved_df.head()

,Date,Close,High,Low,Open,Volume
0,2021-06-09,69.959999,70.620003,69.459999,70.010002,410820
1,2021-06-10,70.290001,70.650002,68.680000,69.760002,492464
2,2021-06-11,70.910004,71.239998,69.680000,70.099998,563575
3,2021-06-14,70.879997,71.779999,70.650002,70.650002,319639
4,2021-06-15,72.120003,72.489998,70.809998,71.160004,336035


In [17]:
df.set_index('Date', inplace=True)

In [18]:
df.head()

Price,Close,High,Low,Open,Volume
Date,,,,,
2021-06-09,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,71.239998,69.680000,70.099998,563575
2021-06-14,70.879997,71.779999,70.650002,70.650002,319639
2021-06-15,72.120003,72.489998,70.809998,71.160004,336035


In [19]:
df.columns

Index(['Close', 'High', 'Low', 'Open', 'Volume'], dtype='str', name='Price')

In [21]:
saved_df.shape

(1258, 6)

In [22]:
df.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 1258 entries, 2021-06-09 to 2026-06-09
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Close   1258 non-null   float64
 1   High    1258 non-null   float64
 2   Low     1258 non-null   float64
 3   Open    1258 non-null   float64
 4   Volume  1258 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 59.0 KB


In [23]:
print(df.columns.tolist())

['Close', 'High', 'Low', 'Open', 'Volume']


In [24]:
df.drop(columns=['index'], inplace=True)

KeyError: "['index'] not found in axis"

In [67]:
df.head()

Price,Close,High,Low,Open,Volume
Date,,,,,
2021-06-07,69.230003,70.000000,68.930000,69.519997,365621
2021-06-08,70.050003,70.269997,68.470001,69.290001,441302
2021-06-09,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,71.239998,69.680000,70.099998,563575


In [68]:
df.head(2)

Price,Close,High,Low,Open,Volume
Date,,,,,
2021-06-07,69.230003,70.000000,68.930000,69.519997,365621
2021-06-08,70.050003,70.269997,68.470001,69.290001,441302


In [70]:
df.columns.name = None

In [71]:
df.head()

,Close,High,Low,Open,Volume
Date,,,,,
2021-06-07,69.230003,70.000000,68.930000,69.519997,365621
2021-06-08,70.050003,70.269997,68.470001,69.290001,441302
2021-06-09,69.959999,70.620003,69.459999,70.010002,410820
2021-06-10,70.290001,70.650002,68.680000,69.760002,492464
2021-06-11,70.910004,71.239998,69.680000,70.099998,563575


In [72]:
df.describe()

,Close,High,Low,Open,Volume
count,1258.000000,1258.000000,1258.000000,1258.000000,1.258000e+03
mean,78.024714,79.503529,76.546256,78.079499,3.201454e+05
std,13.105781,13.730567,12.479617,13.136767,1.269989e+05
min,55.270000,56.700001,54.980000,55.230000,0.000000e+00
25%,69.232502,70.342501,68.165003,69.314999,2.592615e+05
50%,75.829998,77.290001,74.559998,75.875000,3.161315e+05
75%,83.805000,85.547499,82.482498,83.855001,3.778122e+05
max,123.699997,130.500000,120.790001,124.660004,1.107193e+06


In [73]:
df.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 1258 entries, 2021-06-07 to 2026-06-05
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Close   1258 non-null   float64
 1   High    1258 non-null   float64
 2   Low     1258 non-null   float64
 3   Open    1258 non-null   float64
 4   Volume  1258 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 59.0 KB


In [74]:
df.to_csv("../data/processed/wti.csv", index=True)

## Data Collection Summary

- Dataset contains daily WTI Crude Oil Futures prices.
- Covers period from 07-06-2021 to 05-06-2026.
- Contains 1258 trading sessions.
- Includes Open, High, Low, Close, Adj Close, and Volume fields.
- Dataset successfully stored in raw data repository for further analysis.